tests/phpunit/tests/admin/wpTermsListTable.php

```diff
@@ -0,0 +1,125 @@
+
+	private static $admin_id;
+	private static $author_id;
+	private static $term_object;
+
+	const CATEGORY_TAXONOMY = 'category';
+
+	public static function set_up_before_class() {
+		parent::set_up_before_class();
+
+		self::$admin_id  = self::factory()->user->create( array( 'role' => 'administrator' ) );
+		self::$author_id = self::factory()->user->create( array( 'role' => 'author' ) );
+
+		self::$term_object = self::factory()->term->create_and_get( array( 'taxonomy' => self::CATEGORY_TAXONOMY ) );
+
+		require_once ABSPATH . 'wp-admin/includes/class-wp-list-table.php';
+		require_once ABSPATH . 'wp-admin/includes/class-wp-terms-list-table.php';
+	}
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->terms_list_table = new WP_Terms_List_Table();
+	}
+
+	/**
+	 * Call a private method as if it was public.
+	 *
+	 * @param object|string $object      Object instance or class string to call the method of.
+	 * @param string        $method_name Name of the method to call.
+	 * @param array         $args        Optional. Array of arguments to pass to the method.
+	 * @return mixed Return value of the method call.
+	 * @throws ReflectionException If the object could not be reflected upon.
+	 */
+	private function call_private_method( $object, $method_name, $args = array() ) {
+		$method = ( new ReflectionClass( $object ) )->getMethod( $method_name );
+		$method->setAccessible( true );
+		return $method->invokeArgs( $object, $args );
+	}
+
+	/**
+	 * This test proves the existence and reproducibility of the deprecation warnings
+	 * caused by passing null as an argument to `add_query_arg()`
+	 *
+	 * @ticket 59336
+	 *
+	 * @covers WP_Terms_List_Table::handle_row_actions()
+	 */
+	public function test_handle_row_actions_should_generate_deprecation_notice() {
+		if ( PHP_VERSION_ID < 80100 ) {
+			$this->markTestSkipped( 'This test requires PHP 8.1 or higher.' );
+		}
+
+		wp_set_current_user( self::$admin_id );
+
+		$edit_link = add_query_arg(
+			'wp_http_referer',
+			admin_url( 'index.php' ),
+			get_edit_term_link( self::$term_object, self::CATEGORY_TAXONOMY, 'post' )
+		);
+
+		$this->assertStringContainsString( admin_url( 'term.php' ), $edit_link );
+
+		wp_set_current_user( self::$author_id );
+
+		set_error_handler(
+			static function ( $errno, $errstr ) {
+				throw new ErrorException( $errstr, $errno );
+			},
+			E_ALL
+		);
+
+		$this->expectException( ErrorException::class );
+		$this->expectExceptionMessageMatches( '/^strstr\(\): Passing null to parameter #1 \(\$haystack\) of type string is deprecated/' );
+
+		$edit_link = add_query_arg(
+			'wp_http_referer',
+			admin_url( 'index.php' ),
+			get_edit_term_link( self::$term_object, self::CATEGORY_TAXONOMY, 'post' )
+		);
+
+		$this->assertStringNotContainsString( admin_url( 'term.php' ), $edit_link );
+
+		restore_error_handler();
+	}
+
+	/**
+	 * @covers WP_Terms_List_Table::handle_row_actions()
+	 */
+	public function test_handle_row_actions() {
```

Comment:
Not a _must_, but a _nice to have_/best practice recommendations:

1. Split this test in two, one to test with `$author_id`, another to test with `$admin_id`.
    I would suggest a data provider, but in this case that wouldn't be a good solution as the assertions are different for the two situations being tested, so in this case, splitting the test into two separate tests each testing one specific situation is the better solution.
2. Add the `$message` parameter to each assertion.
    When there are multiple assertions in a test, it can sometimes be confusing to see which assertion failed. Using the `$message` parameter you can pass a "failure message" to each assertion to make it more clear cut to see in the command line output, what the expectation was which wasn't met.
    Ref: https://docs.phpunit.de/en/9.6/assertions.html#assertstringcontainsstring

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5198/commits/2c119129e4165e569f8297e2192303149af8a8d7